ide/coqide/coqOps.ml

```diff
@@ -14,7 +14,33 @@ open Ideutils
 open Interface
 open Feedback
 
-let b2c = byte_off_to_buffer_off
+let last_b2c_offsets = ref (0,0)
+
+(** variant of byte_off_to_buffer_off: convert UTF-8 byte offset (used by Coq)
+    to unicode offset (used by GTK buffer) with size caching for performance *)
+let b2c buffer byte_off upd =
+  if byte_off < 0 then
+    raise (invalid_arg "b2c: byte_off must be >= 0");
+  let (s_uni, s_byte) = if byte_off >= (snd !last_b2c_offsets) then !last_b2c_offsets else (0,0) in
+  let rec cvt iter cnt =
+    if cnt <= 0 then
+      iter#offset
+    else
+      cvt iter#forward_char (cnt - (ulen iter#char))
+  in
+  let uni_off = cvt (buffer#get_iter (`OFFSET s_uni)) (byte_off - s_byte) in
+  if upd && byte_off > s_byte + 50 then
+    last_b2c_offsets := (uni_off, byte_off);
+  uni_off
+
+(* convert byte offset to char offset in a Loc.t *)
+let loc_to_uni (buffer : GText.buffer) loc =
```

Comment:
Don't reuse the same type to encode locations with UTF8-based characters. Either create a new record type, or simply return a pair of integers. The first thing that is done with it any way is to convert it to such a pair immediately.

Reply:
Done